CodeEdit/Features/QuickOpen/ViewModels/FuzzySearch.swift

```diff
@@ -0,0 +1,207 @@
+            return nil
+        }
+    }
+
+    /// Calculate the date score for a given file's modification date.
+    ///
+    /// This function calculates the date score based on the time difference
+    /// between the current date and the file's modification date,
+    /// using an exponential decay function with a half-life of 3600 seconds (1 hour).
+    /// The score will be higher for more recently modified files.
+    ///
+    /// - Parameter modificationDate: The file's modification date.
+    /// - Returns: The date score as a `Double` value.
+    private static func calculateDateScore(for modificationDate: Date) -> Double {
+        let now = Date()
+        let timeDiff = now.timeIntervalSince(modificationDate)
+        let halfLife: Double = 3600 // decay half-life in seconds
+        let decayFactor = log(2) / halfLife
+        let score = exp(-decayFactor * timeDiff)
+        return score
+    }
+
+    /// Calculates the Levenshtein distance between two input strings.
+    ///
+    /// - Parameters:
+        /// - sourceString: The source string to compare against the target string;
+        /// - targetString: The target string to compare against the source string.
+    /// - Returns: The Levenshtein distance between `sourceString` and `targetString`.
+    /// An integer representing the minimum number of
+    ///  insertions, deletions, or substitutions required to transform the source string into the target string.
+    private static func levenshteinDistance(from sourceString: String, to targetString: String) -> Int {
+        let source = Array(sourceString)
+        let target = Array(targetString)
+
+        let sourceCount = source.count
+        let targetCount = target.count
+
+        guard sourceCount > 0 else {
+            return targetCount
+        }
+
+        guard targetCount > 0 else {
+            return sourceCount
+        }
+
+        var distanceMatrix = Array(repeating: Array(repeating: 0, count: targetCount + 1), count: sourceCount + 1)
+
+        for rowIndex in 0...sourceCount {
+            distanceMatrix[rowIndex][0] = rowIndex
+        }
+
+        for columnIndex in 0...targetCount {
+            distanceMatrix[0][columnIndex] = columnIndex
+        }
+
+        for rowIndex in 1...sourceCount {
+            for columnIndex in 1...targetCount {
+                let cost = source[rowIndex - 1] == target[columnIndex - 1] ? 0 : 1
+
+                let deletionCost = distanceMatrix[rowIndex - 1][columnIndex] + 1
+                let insertionCost = distanceMatrix[rowIndex][columnIndex - 1] + 1
+                let substitutionCost = distanceMatrix[rowIndex - 1][columnIndex - 1] + cost
+
+                distanceMatrix[rowIndex][columnIndex] = min(deletionCost, insertionCost, substitutionCost)
+            }
+        }
+
+        return distanceMatrix[sourceCount][targetCount]
+    }
+}
+
+extension String {
+    /// Adds a new function to the `String` type that searches for all occurrences
```

Comment:
No need for the "Adds a new" portion of the comment. You can simply describe what the method does.